### DISCUSSION TITLE
Feature Request: enable EXCLUDE using COLUMNS

### DISCUSSION THREAD

**Discussion by aborruso** [duckdb/duckdb#8969] [Ideas]
Sometimes we must exclude a lot of columns, which have some pattern in their name.

It would be useful to be able to exclude them using `COLUMNS`: `select * EXCLUDE(COLUMNS("^[a-v].+")) from input`.

Thank you

**Comment by liquidcarbon** [comment]
That could be handy on the "include" side as well, i.e. `SELECT COLUMNS(matching regex)...`

**Reply by aborruso** [reply]
> That could be handy on the "include" side as well, i.e. `SELECT COLUMNS(matching regex)...`

this already works

**Comment by mskyttner** [comment]
+1 for this but perhaps with a wider scope (not only for excluding certain column names, but more generally in duck-sql to support more "selection helpers" when working with column names).

For column-oriented data, to be able to treat column names as data ("dynamic aliasing" etc) is very convenient, but such operations are maybe not well supported in traditional sql. The need seems to become even stronger when introducing pivoting/unpivoting support. So perhaps there are opportunities for ducksql to support even friendlier sql when working with column names...? 

I think the API offered through the tidyselect R package offers [a lot of useful convenience in this area](https://tidyselect.r-lib.org/reference/language.html). It provides a small DSL which supports selecting columns based on different conditions, such as explicit include/exclude lists of column names (just negate the condition for exclude), patterns (including negated patterns) and even based on the data type of the columns. The "all_of" or "any_of" feature provides a way to indicate to the query engine whether an error should occur when for example selecting a column name which does not exist in the data.

**Comment by Tishj** [comment]
I think all of this functionality can be achieved through lambdas in the COLUMNS, added in this PR:
<https://github.com/duckdb/duckdb/pull/6621>

I don't think we have any plans to extend this further
That does not include the aliasing problem that has come up quite a bunch, I personally think that should be addressed

**Comment by Tishj** [comment]
> Sometimes we must exclude a lot of columns, which have some pattern in their name.
> 
> It would be useful to be able to exclude them using `COLUMNS`: `select * EXCLUDE(COLUMNS("^[a-v].+")) from input`.
> 
> Thank you

Exclude cant take COLUMNS, but COLUMNS itself can already be used to create a whitelist/blacklist, so exclude is not necessary here

**Comment by aborruso** [comment]
> COLUMNS itself can already be used to create a whitelist/blacklist

Thank you @Tishj
